lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -147,7 +148,14 @@ def _put(self, params: ApiParams, _) -> Response:
             for k in metadata.keys():
                 if Metadata.get_native_key(k) != Metadata.USER:
                     role = API_OPERATION.UPDATE
-        self._check_authorization(str(dataset.owner_id), dataset.access, role)
+        self._check_authorization(
+            ApiAuthorization(
+                API_AUTHORIZATION.USER_ACCESS,
+                role,
+                str(dataset.owner_id),
+                dataset.access,
+            )
```

Comment:
Given that `ApiAuthorization` is a `NamedTuple`, would it be appropriate to use keyword parameters?

Reply:
Takes a lot more space, and I was already blowing these up by consolidating the parameters into the `ApiAuthorization` (it was that or add another parameter everywhere, and I remembered that you had suggested this earlier). Anyway, I was just conserving space and editing time ... but I'll think about it.